inference-engine/src/transformations/src/transformations/common_optimizations/pad_fusion.cpp

```diff
@@ -0,0 +1,202 @@
+    auto node_pads_begin = node->get_pads_begin();
+    auto node_pads_end = node->get_pads_end();
+    auto pads_begin_val = pads_begin->cast_vector<size_t>();
+    auto pads_end_val = pads_end->cast_vector<size_t>();
+
+    std::transform(node_pads_begin.begin(), node_pads_begin.end(),
+                   pads_begin_val.begin() + 2, node_pads_begin.begin(),
+                   [] (size_t a, size_t b) -> size_t { return a + b; });
+    std::transform(node_pads_end.begin(), node_pads_end.end(),
+                   pads_end_val.begin() + 2, node_pads_end.begin(),
+                   [] (size_t a, size_t b) -> size_t { return a + b; });
+
+    return std::make_tuple(node_pads_begin, node_pads_end);
+}
+
+NGRAPH_RTTI_DEFINITION(pass::PadFusionAvgPool, "PadFusionAvgPool", 0);
+
+pass::PadFusionAvgPool::PadFusionAvgPool() {
+    auto data_pattern = pattern::any_input();
+    auto pads_begin_pattern = pattern::wrap_type<opset5::Constant>();
+    auto pads_end_pattern = pattern::wrap_type<opset5::Constant>();
+    auto pad_value_pattern = pattern::wrap_type<opset5::Constant>();
+    auto pad_node_pattern = pattern::wrap_type<opset5::Pad>({data_pattern, pads_begin_pattern, pads_end_pattern, pad_value_pattern});
+    auto avg_pool_pattern = pattern::wrap_type<opset5::AvgPool>({pad_node_pattern});
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        auto pattern_map = m.get_pattern_value_map();
+        auto data = pattern_map[data_pattern];
+        auto pad = std::dynamic_pointer_cast<opset5::Pad>(pattern_map[pad_node_pattern].get_node_shared_ptr());
+        auto pad_value_const = std::dynamic_pointer_cast<opset5::Constant>(pattern_map[pad_value_pattern].get_node_shared_ptr());
+        auto pads_begin = std::dynamic_pointer_cast<opset5::Constant>(pattern_map[pads_begin_pattern].get_node_shared_ptr());
+        auto pads_end = std::dynamic_pointer_cast<opset5::Constant>(pattern_map[pads_end_pattern].get_node_shared_ptr());
+        auto avg_pool = std::dynamic_pointer_cast<opset5::AvgPool>(pattern_map[avg_pool_pattern].get_node_shared_ptr());
+        if (!can_be_fused(pad, avg_pool, pad_value_const, pads_begin, pads_end))
+            return false;
+
+        Shape new_pads_begin, new_pads_end;
+        std::tie(new_pads_begin, new_pads_end) = new_pad_values(pads_begin, pads_end, avg_pool);
+        auto new_avg_pool = std::make_shared<opset5::AvgPool>(data, avg_pool->get_strides(),
+                                                              new_pads_begin, new_pads_end,
+                                                              avg_pool->get_kernel(), false,
+                                                              avg_pool->get_rounding_type(),
+                                                              op::PadType::EXPLICIT);
+        new_avg_pool->set_friendly_name(avg_pool->get_friendly_name());
+
+        copy_runtime_info(avg_pool, new_avg_pool);
```

Comment:
You need to copy rt info from pad to new avg_pool also.
```suggestion
        copy_runtime_info({pad, avg_pool}, new_avg_pool);
```